### PR TITLE
config: cleanup and pass config as parameter

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### CODE REVIEW

File: rpc/client.go [hunk 1]
```diff
@@ -142,32 +144,33 @@ func (c *batchCommandsClient) batchRecvLoop() {
...
 	}
 }
 
-func newConnArray(maxSize uint, addr string, security config.Security) (*connArray, error) {
+func newConnArray(addr string, conf *config.RPC) (*connArray, error) {
 	a := &connArray{
+		conf:                 conf,
 		index:                0,
-		v:                    make([]*grpc.ClientConn, maxSize),
+		v:                    make([]*grpc.ClientConn, conf.MaxCallMsgSize),
```
Comment (rleungx) [inline rpc/client.go]: Maybe this variable name is not good?

File: txnkv/store/txn_committer.go [hunk 2]
```diff
@@ -183,10 +186,10 @@ func (c *TxnCommitter) doActionOnKeys(bo *retry.Backoffer, action commitAction,
 		atomic.AddInt32(&c.detail.PrewriteRegionNum, int32(len(groups)))
 	}
 	// Make sure the group that contains primary key goes first.
-	batches = appendBatchBySize(batches, firstRegion, groups[firstRegion], sizeFunc, config.TxnCommitBatchSize)
+	batches = appendBatchBySize(batches, firstRegion, groups[firstRegion], sizeFunc, c.conf.Txn.CommitBatchSize)
```
Comment (rleungx) [inline txnkv/store/txn_committer.go]: For such kind of things, how about using a variable to make it look more concise?

File: txnkv/kv/memdb_buffer.go [hunk 3]
```diff
@@ -43,12 +42,15 @@ type memDbIter struct {
 }
 
 // NewMemDbBuffer creates a new memDbBuffer.
-func NewMemDbBuffer(cap int) MemBuffer {
+func NewMemDbBuffer(conf *config.Txn, cap int) MemBuffer {
+	if cap == 0 {
```
Comment (rleungx) [inline txnkv/kv/memdb_buffer.go]: How about `<=`?

File: config/raw.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 PingCAP, Inc.
```
Comment (rleungx) [inline config/raw.go]: Do we need to change it to 2019?

File: txnkv/kv/memdb_buffer.go (hunk 3 above)
Comment (disksing) [reply txnkv/kv/memdb_buffer.go]: ok.

File: config/raw.go (hunk 4 above)
Comment (disksing) [reply config/raw.go]: ok.

File: txnkv/store/txn_committer.go (hunk 2 above)
Comment (disksing) [reply txnkv/store/txn_committer.go]: ok.

File: rpc/client.go (hunk 1 above)
Comment (disksing) [reply rpc/client.go]: How about `GrpcMaxCallMsgSize`?

Comment (rleungx) [reply rpc/client.go]: Hmmm...Actually, I mean the `v` here.

**Pull request merged**